drbert/data/dataset_readers.py

```diff
@@ -0,0 +1,416 @@
+            batch_first=True,
+            pad_token=self.tokenizer.pad_token_id,
+            unk_token=self.tokenizer.unk_token_id
+        )
+        self.LABEL = data.LabelField()
+
+    def textual_to_iterator(self, fields):
+        # Define the splits. Need path to directory (self.path), and then name of each file.
+        splits = data.TabularDataset.splits(
+            path=self.path,
+            **self.partitions,
+            format=self.format,
+            fields=fields,
+            skip_header=self.skip_header
+        )
+
+        # Define the iterator. Batch sizes are defined per partition.
+        # BucketIterator groups sentences of similar length together to minimize padding.
+        iterators = data.BucketIterator.splits(
+            splits, batch_sizes=self.batch_sizes, sort_key=self.sort_key
+        )
+
+        return splits, iterators
+
+
+class DocumentClassificationDatasetReader(DatasetReader):
+    """Reads instances from JSON lines file(s) were each line is in the following format:
+
+    {"TEXT": DOCUMENT TEXT, "LABEL": label (if single label task) OR "LABEL_1": label_1, "LABEL_2: label_2, ..., "LABEL_N": label_n (if n-labeled multi-label task)
+
+    Example usage:
+        >>> from transformers import AutoTokenizer
+        >>> from dataset_readers import DocumentClassificationDatasetReader
+        >>> partitions={
+                'train': 'train_filename.jsonl',
+                'validation': 'valid_filename.jsonl',
+                'test': 'test_filename.jsonl',
+            }
+        >>> tokenizer = AutoTokenizer.from_pretrained('bert-base-cased')
+        >>> train_iter, valid_iter, test_iter = DocumentClassificationDatasetReader(
+                path='path/to/nli/dataset', partitions=partitions, tokenizer=tokenizer,
+                batch_sizes=(16, 256, 256)
+            ).textual_to_iterator()
+
+    Args:
+        path (str): Common prefix of the splits’ file paths.
+        partitions (dict): A dictionary keyed by partition ('train', 'test', 'validation')
+            containing the suffix to add to `path` for that partition.
+        tokenizer (transformers.PreTrainedTokenizer): The function used to tokenize strings into
+            sequential examples.
+        batch_sizes (tuple): Optional, tuple of batch sizes to use for the different splits, or None
+            to use the same batch_size for all splits. Defaults to None.
+        lower: (bool): Optional, True if text should be lower cased, False if not. Defaults to
+            False.
+        multi_label_map (dict or list): dict or list containing all fields in the label space as per the jsonl file (for multi-label document classification only). If None, will assume single label document classification. Defaults to None 
+    """
+    def __init__(self, path, partitions, tokenizer, batch_sizes=None, lower=False, multi_label_map=None):
+        super(DocumentClassificationDatasetReader, self).__init__(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Bad first argument 'DocumentClassificationDatasetReader' given to super()](https://app.codacy.com/app/JohnGiorgi/drbert/pullRequest?prid=4302662)